### PR TITLE
Document privatelink.endpoint option for Kafka source/sink

### LINKED ISSUE #1329: Document: feat: support `privatelink.endpoint` option in the WITH clause for Kafka connector
This issue tracks the documentation update needed for the merged PR #12266.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/12266
Source PR Merged At: 2023-09-22T01:49:52Z

### PR DESCRIPTION
**Pull request by CharlieSYH**
## Info

- **Description**

  Document `privatelink.endpoint` option in the WITH clause in Kafka source/sink connector

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/12266

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1329

## For reviewers

- **Preview**

  https://pr-1341.d2fbku9n2b6wde.amplifyapp.com/docs/dev/create-source-kafka/#create-source-with-aws-privatelink-connection
 https://pr-1341.d2fbku9n2b6wde.amplifyapp.com/docs/dev/create-sink-kafka/#create-sink-with-aws-privatelink-connection

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Review by emile-00 — APPROVED** [review]

File: docs/create-source/create-source-kafka.md [hunk 1]
```diff
@@ -372,25 +372,35 @@ To create a Kafka source with a PrivateLink connection, in the WITH section of y
 
 |Parameter| Notes|
 |---|---|
-|`connection.name`| The name of the connection, which comes from the connection created using the `CREATE CONNECTION` statement.|
+|`connection.name`| The name of the connection, which comes from the connection created using the [`CREATE CONNECTION`](/sql/commands/sql-create-connection.md) statement.|
 |`privatelink.targets`| The PrivateLink targets that correspond to the Kafka brokers. The targets should be in JSON format. Note that each target listed corresponds to each broker specified in the `properties.bootstrap.server` field. If the order is incorrect, there will be connectivity issues. |
+|`privatelink.endpoint`|The DNS name of the AWS VPC endpoint or the GCP private link endpoint.|
```
Comment (StrikeW) [inline docs/create-source/create-source-kafka.md]: We need to remind users that `privatelink.endpoint` is a recommended way to use privatelink, which means user don't need to create a connection in RW if they have a provisioned privatelink endpoint.

File: docs/guides/create-sink-kafka.md [hunk 2]
```diff
@@ -154,6 +155,7 @@ WITH (
    topic='msk_topic',
    force_append_only='true',
    connection.name = 'connection1',
+   privatelink.endpoint='10.148.0.4',
```
Comment (StrikeW) [inline docs/guides/create-sink-kafka.md:157]: This is incorrect, `privatelink.endpoint` and `connection.name` are two different way to use privatelink, and the former one is recommended. The create connection way only supported in AWS.

File: docs/create-source/create-source-kafka.md (hunk 1 above)
Comment (CharlieSYH) [reply docs/create-source/create-source-kafka.md]: Updated. Please review again, thanks!

File: docs/guides/create-sink-kafka.md (hunk 2 above)
Comment (CharlieSYH) [reply docs/guides/create-sink-kafka.md:157]: Dropped `connection.name = 'connection1',`

File: docs/create-source/create-source-kafka.md [hunk 3]
```diff
@@ -372,9 +372,9 @@ To create a Kafka source with a PrivateLink connection, in the WITH section of y
...
 |`privatelink.targets`| The PrivateLink targets that correspond to the Kafka brokers. The targets should be in JSON format. Note that each target listed corresponds to each broker specified in the `properties.bootstrap.server` field. If the order is incorrect, there will be connectivity issues. |
-|`privatelink.endpoint`|The DNS name of the AWS VPC endpoint or the GCP private link endpoint.|
+|`privatelink.endpoint`|The DNS name of the AWS VPC endpoint or the GCP Private Service Connect endpoint. <br/> Using `privatelink.endpoint` is the recommended way to use PrivateLink. With a provisioned PrivateLink endpoint, you don't need to create a connection in RisingWave.|
+|`connection.name`| The name of the connection. <br/> THis parameter should only be included if you are using a AWS PrivateLink connection created with the [`CREATE CONNECTION`](/sql/commands/sql-create-connection.md) statement. Omit this parameter if you have provisioned an AWS VPC endpoint or the GCP Private Service Connect endpoint using `privatelink.endpoint` (recommended).|
```
Comment (StrikeW) [inline docs/create-source/create-source-kafka.md]: THis -> This

**Review by StrikeW — APPROVED** [review]
Rest lgtm.

**Pull request merged**